de1plus/plugins/webhook/plugin.tcl

```diff
@@ -0,0 +1,129 @@
+
+namespace eval ::plugins::${plugin_name} {
+
+    variable author "Kevin Gao"
+    variable contact "@sudowork"
+    variable version 1.0
+    variable description "Sends shot data as a webhook to a specified URL"
+    variable name "Webhook"
+
+    proc preload {} {
+        return [build_settings_ui]
+    }
+
+    proc build_settings_ui {}  {
+        # Create settings if non-existant
+        if {[array size ::plugins::webhook::settings] == 0} {
+            array set ::plugins::webhook::settings {
+                webhook_domain "example.com"
+                webhook_endpoint "/decent/webhook"
+                webhook_secret_key "SecretKeyForSigning"
+                webhook_trigger_after 10
+            }
+            plugins save_settings webhook
+        }
+
+        # Unique name per page
+        set page_name "plugin_webhook_page_default"
+
+        # Background image and "Done" button
+        add_de1_page "$page_name" "settings_message.png" "default"
+        add_de1_text $page_name 1280 1310 -text [translate "Done"] -font Helv_10_bold -fill "#fAfBff" -anchor "center"
+        add_de1_button $page_name {say [translate {Done}] $::settings(sound_button_in); page_to_show_when_off extensions}  980 1210 1580 1410 ""
+
+        # Headline
+        add_de1_text $page_name 1280 300 -text [translate "Webhook"] -font Helv_20_bold -width 1200 -fill "#444444" -anchor "center" -justify "center"
+
+        # Render Settings
+        build_setting_text_input $page_name [translate "Webhook Domain"] {::plugins::webhook::settings(webhook_domain)} 480
+        build_setting_text_input $page_name [translate "Webhook Endpoint"] {::plugins::webhook::settings(webhook_endpoint)} 660
+        build_setting_text_input $page_name [translate "Webhook Secret Key"] {::plugins::webhook::settings(webhook_secret_key)} 840
+        build_setting_text_input $page_name [translate "Trigger After (seconds)"] {::plugins::webhook::settings(webhook_trigger_after)} 1020
+
+        return $page_name
+    }
+
+    proc build_setting_text_input {page_name label var y} {
+        set x 280
+        add_de1_text $page_name $x $y -text $label -font Helv_8 -width 300 -fill "#444444" -anchor "nw" -justify "center"
+        add_de1_widget $page_name entry $x [expr $y + 60]  {
+            bind $widget <Return> { say [translate {save}] $::settings(sound_button_in); borg toast [translate "Saved"]; save_plugin_settings webhook; hide_android_keyboard}
+            bind $widget <Leave> hide_android_keyboard
+        } -width [expr {int(38 * $::globals(entry_length_multiplier))}] -font Helv_8  -borderwidth 1 -bg #fbfaff  -foreground #4e85f4 -textvariable "$var" -relief flat  -highlightthickness 1 -highlightcolor #000000
+    }
+
+    proc post_shot_data {} {
+        variable settings
+
+        msg "triggering webhook"
+        borg toast [translate "Triggering Webhook"]
+
+        http::register https 443 [list ::tls::socket -servername $settings(webhook_domain)]
```

Comment:
This may be my misunderstanding, but I don’t believe this is binding anything. Based on my read of the docs, it’s used to establish a TLS transport for making the HTTPS request to the server: https://www.tcl-lang.org/man/tcl/TclCmd/http.htm#M49